casper/src/main/scala/coop/rchain/casper/api/BlockAPI.scala

```diff
@@ -174,7 +166,7 @@ object BlockAPI {
       val stateHash =
         ProtoUtil.tuplespace(block).get
       val data =
-        runtimeManager.getData(stateHash, sortedListeningName)
+        runtimeManager.getData(stateHash, sortedListeningName).unsafeRunSync
```

Comment:
Instead of doing `unsafeRunSync` here, add a type constraint of `ToAbstractContext` to the function's signature, lift the task to `F` and move this line into the for comprehension below